File: java01/src/step28/ex1/MemberDao.java
package step28.ex1;
/* 역할:
 * => memb 테이블의 데이터를 다루는 메서드를 모아둔 클래스이다.
 * => 출력하는 역할은 호출자에게 맡긴다.
 * => DAO의 역할은 결과를 리턴하면 된다.
 * => 커넥션은 DBConnectionPool로부터 얻어서 사용하고,
 *      사용한 후에는 반납한다.
 */


import java.util.List;


public interface MemberDao {
  List<Member> selectList(int pageNo, int pageSize) throws Exception;
  Member selectOne(int no) throws Exception;
  int insert(Member member) throws Exception;
  int delete(int no) throws Exception;
  int update(Member member) throws Exception;
  Member selectOneByEmailPassword (String email, String password) throws Exception;
}

File: electron01/step11/member-dao.js
// 데이터 처리를 담당하는 자바스크립트
// => DAO(Data Access Object)
"use strict"

function createMemberDao(connection) {
  return {
      "insert" : function(member, successFn, errorFn) {
      connection.query(
        'insert into memb(name,tel,email,pwd) values(?,?,?,password(?))',
        [ member.name, member.tel, member.email, member.password],
        function(error, result) {
          if (error) {
            errorFn(error)
          } else {
            successFn(result)
          }
        }) //connection.query()
      }, //insertStudent()
      ///////////////////////////////////////////////////////
      "update" : function (member, successFn, errorFn) {
        connection.query(
          'update memb set name=?, tel=?, email=? where mno =?',
          [member.name, member.tel, member.email, member.no],
          function(error, result) {
            if (error) {
              alert('회원 기본 데이터 변경 중 오류 발생!')
              throw error
            } else {
              successFn(result)
            }
          })
        }, // updateMember
///////////////////////////////////////////////
      "delete": function(no , successFn, errorFn) {
        connection.query(
          'delete from memb where mno = ?',
          [no],
          function(error, result) {
            if(error) {
              errorFn(error)
            } else {
              successFn(result)
            }
          })
        }//delete
      //////////////////////////////////////////
  } //return
}
